accepted/2020/binary-data/binary-data.md

```diff
@@ -0,0 +1,177 @@
+struggling to find the right method, or even any, to produce the data from the
+data they already have.
+
+## Scenarios and User Experience
+
+### Sending a binary payload
+
+Naomi is building a cloud-based solution for batch resizing of images. She uses
+Azure Storage Queues to manage requests for image resizing. In order to queue a
+message, she uses the `SendMessageAsync` API. The method takes a type called
+`BinaryData`. Curious what this is, she decides to new one up. Since there is no
+constructor, she explores the static members. Naomi realizes that there are
+several `FromXxx` methods. "Hmm, I need to serialize my object somehow", she
+concludes that `FromObjectAsJson` is the right API for her.
+
+```C#
+Task QueueImageResize(string imageUrl)
+{
+    var queueClient = new QueueClient(userSecret.QueueConnectionString, "resize-queue");
+
+    var message = new ResizeImageMessage
+    {
+        Url = imageUrl,
+        Resolutions = new[] {"200x200", "400x400", "1200x1200" }
+    };
+
+    var messageData = BinaryData.FromObjectAsJson(message);
+    await queueClient.SendMessageAsync(messageData);
+}
+```
+
+### Retrieving a binary payload
+
+Shirin works on Naomi's team but is responsible for the resizing backend. She
+wrote an Azure Function that is triggered every time an item is added to the
+queue. The template already created a handler `RunAsync` that passes the queue
+item as an instance of `BinaryData`.
+
+In order to deserialize, she simply invokes the `ToObjectFromJson` method:
+
+```C#
+Task RunAsync(BinaryData messageData)
+{
+    var message = messageData.ToObjectFromJson<ResizeImageMessage>();
+
+    foreach (var resolution in message.Resolutions)
+    {
+        var newUrl = GetUrl(resolution.Url, resolution);
+        var image = await DownloadImage(resolution.Url);
+        image.Resize(resolution);
+        await UploadImageAsync(newUrl, image);
+    }
+}
+```
+
+## Requirements
+
+### Goals
+
+* Stable API
+* Make it easy for people to construct bytes from common data types (byte
+  arrays, spans, memories, string, JSON serialized objects)
+* Make it easy for people to convert bytes to common data types of the same
+  representations
+* Provide opinionated defaults in order to reduce complexity
+    - String encoding is assumed to be UTF8
+    - Object serialization is done with `System.Text.Json`, with specific options
+
+### Non-Goals
+
+* Allocation free zero-copying semantics. The primary goal is convenience and
+  usability, not perf. Performance critical scenarios should be enabled by
+  providing overloads with the existing low-level concepts, such as span.
+
+## Design
+
+We're concerned with hard coding the type to `System.Text.Json`, but we
+acknowledge that defaults are very valuable. If we need to support other
+serializers we can provide additional arguments or APIs.
+
+This type can't live in corlib because it would depend on `System.Text.Json`
```

Comment:
If it can't be a part of foundational APIs, then why are we placing it in `System.*` namespace?

Reply:
Being in corlib isn't the same as being foundational. `System.Text.Json` isn't in corlib either.